web/packages/shared/utils/highbar.ts

```diff
@@ -0,0 +1,273 @@
+/**
+ * Copyright 2023 Gravitational, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export function mergeDeep(target, ...sources) {
```

Comment:
Can we add types to all functions?

Reply:
can we also add tests for this file? probably can skip ones that we directly copied from lodash

i did a quick test for `mergeDeep`, and it didn't deep copy an array inside of object and i believe lodash handles this, should we just copy this from lodash too?

Reply:
This is correct, it only merges objects which is all we need for the single usecase in the app.

The lodash version is hundreds of lines long and extensively covers all of the different JS datatypes.

Reply:
I was contemplating removing the necessity of this util at all but I didn't really want to scope creep this task as it was.

Reply:
Turns out that this merge fn introduces a bug because the config value that gets merged can have nested arrays of objects so I'll have to update this to support arrays as well.

Reply:
This fn now supports arrays as well.